kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -969,15 +1017,20 @@ This eliminates any `Deref` projections from the place, and also resolves `Index
 
   // once traversal is finished, reconstruct the last projections and the reference offset/local, and possibly read the size
   rule <k> #traverseProjection(DEST, VAL:Value, .ProjectionElems, CTXTS) ~> #forRef(MUT, META)
-        => #mkRef(DEST, #projectionsFor(CTXTS), MUT, #maybeDynamicSize(META, VAL))
+        => #mkRef(DEST, #projectionsFor(CTXTS), MUT, #maybeDynamicSize(META, VAL), 0) // TODO This needs to be changed
         ...
```

Comment:
Agree the zero is not what we want here. 
There is _probably_ an invariant that offset projection contexts will always occur at the start of the context (or at the end? :thinking: does not make sense, so maybe this decides the `#derefTruncate` must happen before the offset ). `mkRef` can then match that context and extract the correct offset value.
(This is all not exact advice, just thoughts)